ckb-bin/src/subcommand/db_repair.rs

```diff
@@ -1,7 +1,13 @@
+use crate::helper::raise_fd_limit;
 use ckb_app_config::{ExitCode, RepairArgs};
 use ckb_db::RocksDB;
+use ckb_logger::info;
 
 pub fn db_repair(args: RepairArgs) -> Result<(), ExitCode> {
+    if let Some(limit) = raise_fd_limit() {
```

Comment:
suggest to move `raise_fd_limit` code to `run_app`:
https://github.com/nervosnetwork/ckb/blob/fc94d8c8468b676000a33fa8dc415ac1761a86d5/ckb-bin/src/lib.rs#L48

Reply:
Maybe, the sub-commands without using `RocksDB` don't have to raise `fd_limit`.

Like apps in Android, no one likes those apps which always request unnecessary permissions.

Reply:
> Maybe, the sub-commands without using RocksDB don't have to raise fd_limit.
Yes, that's the reason of suggestion to move this code to line 48: before line 48, the subcommands are unrelated to rocksdb.

Reply:
> > Maybe, the sub-commands without using RocksDB don't have to raise fd_limit.
> 
> Yes, that's the reason of suggestion to move this code to line 48: before line 48, the subcommands are unrelated to rocksdb.

I think you misunderstanding what @yangby-cryptape said. if this code moves to `run_app` line 48, all subcommands will require raise fd_limit, which he against.

Reply:
the sub-commands init / list_hashes / etc will return early:
https://github.com/nervosnetwork/ckb/blob/fc94d8c8468b676000a33fa8dc415ac1761a86d5/ckb-bin/src/lib.rs#L30-L41

Reply:
> after line 48, all sub-commands run / migrate / export / etc will open rocksdb.

Nope! :stuck_out_tongue_closed_eyes: For example, the line 60:
https://github.com/nervosnetwork/ckb/blob/fc94d8c8468b676000a33fa8dc415ac1761a86d5/ckb-bin/src/lib.rs#L60

Reply:
We may move this subcommand before raise_fd_limit